rasa/cli/x.py

```diff
@@ -291,16 +304,76 @@ def rasa_x(args: argparse.Namespace):
         run_locally(args)
 
 
-def run_in_production(args: argparse.Namespace):
-    from rasa.cli.utils import print_success
-    from rasa.core.utils import AvailableEndpoints
+def _pull_runtime_config_from_server(
+    config_endpoint: Optional[Text],
+    attempts: int = 30,
+    wait_time_between_pulls: Union[int, float] = 0.5,
+    keys: Tuple[Text, ...] = ("endpoints", "credentials"),
+) -> Tuple[Text, ...]:
+    """Pull runtime config from `config_endpoint`.
 
-    print_success("Starting Rasa X in production mode... 🚀")
+    Returns a path to yaml dumps, each containing the contents of one of `keys`.
+    """
+
+    while attempts > 0:
+        try:
+            response = requests.get(config_endpoint)
```

Comment:
Mhm, does not really matter at startup whether we use `asyncio` or not but on the other hand it's the only point in `rasa` code where we use it (except one channel) 🤔